src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs

```diff
@@ -142,7 +142,8 @@ private static bool TokenShouldNotFormatOnReturn(SyntaxToken token)
 
         private static bool TokenShouldNotFormatOnTypeChar(SyntaxToken token)
         {
-            return token.IsKind(SyntaxKind.CloseParenToken) && !token.Parent.IsKind(SyntaxKind.UsingStatement);
+            return token.IsKind(SyntaxKind.CloseParenToken) && !token.Parent.IsKind(SyntaxKind.UsingStatement) ||
```

Comment:
Can you add parens around the appropriate clauses?  It'll be a lot easier to read and modify.